implementations/rust/ockam/ockam_command/src/base_command.rs

```diff
@@ -524,86 +525,139 @@ impl Display for ReadStatus {
     }
 }
 
-struct StdinHandler {}
+struct RustylineHandler {}
 
-struct StdinHandle {
-    tx: tokio::sync::broadcast::Sender<String>,
+struct RustylineHandle {
+    lines: Receiver<String>,
+    next: Sender<()>,
 }
 
-impl StdinHandler {
-    async fn read_input(
-        stdin: &mut BufReader<tokio::io::Stdin>,
-        read_buffer: &mut String,
-    ) -> std::io::Result<usize> {
-        let n = stdin.read_line(read_buffer).await?;
-
-        let separator = "\"\"\"\n";
-        if read_buffer == separator {
-            loop {
-                stdin.read_line(read_buffer).await?;
-
-                if read_buffer.ends_with(separator) {
-                    break;
-                }
+impl RustylineHandle {
+    async fn read_line(&mut self) -> Option<String> {
+        match self.next.send(()).await {
+            Ok(_) => self.lines.recv().await.map(|line| line + "\n"),
+            Err(_) => {
+                // Channel closed, exit REPL
+                None
             }
         }
-
-        Ok(n)
     }
+}
 
-    fn start() -> StdinHandle {
-        let (tx, _) = tokio::sync::broadcast::channel(64);
-        let returned_tx = tx.clone();
-
-        // Spawn a task to read from stdin
-        tokio::spawn(async move {
-            // Ctrl+C handler to exit stdin loop
-            let (cancel_tx, cancel_rx) = tokio::sync::broadcast::channel::<()>(1);
-            let mut processed = false;
-            ctrlc::set_handler(move || {
-                if !processed {
-                    let _ = cancel_tx.send(());
-                    processed = true;
-                }
-            })
-            .expect("Error setting Ctrl+C handler");
-
-            let mut stdin = BufReader::new(tokio::io::stdin());
-            let mut read_buffer = String::new();
+#[derive(Hinter, Completer, Highlighter)]
+struct MultiLineValidatorHelper {}
+impl Validator for MultiLineValidatorHelper {
+    fn validate(
+        &self,
+        ctx: &mut ValidationContext,
+    ) -> std::result::Result<ValidationResult, ReadlineError> {
+        let input = ctx.input();
+        if input.starts_with("\"\"\"") {
```

Comment:
If we match the ending with `"\n\"\"\""`, should we then match the beginning with `"\"\"\"\n"`?